bumble/smp.py

```diff
@@ -946,14 +945,7 @@ def display_passkey(self) -> None:
             self.tk = self.passkey.to_bytes(16, byteorder='little')
             logger.debug(f'TK from passkey = {self.tk.hex()}')
 
-        try:
-            utils.cancel_on_event(
-                self.connection,
-                'disconnection',
-                self.pairing_config.delegate.display_number(self.passkey, digits=6),
-            )
-        except Exception as error:
-            logger.warning(f'exception while displaying number: {error}')
+        await self.pairing_config.delegate.display_number(self.passkey, digits=6)
```

Comment:
nit: This will become blocking, but it looks intended. Maybe we should have a better documentation about which methods are and should be blocking.